File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        zero_count = 0
        one_count = s.count("1")
        answer = 0

        for i in range(len(s) - 1):
            if(s[i] == '0'):
                zero_count += 1
            else:
                one_count -= 1
            
            answer = max(answer,  zero_count + one_count)
        
        return answer

File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int right_zero = 0;
        int right_one = 0;
        int left_zero = 0;
        int left_one = 0;
        for(int i = 0; i < s.length();  i++) {
            if(s.charAt(i) == '0') {
                right_zero++; // 2
            } else {
                right_one++; // 3
            }
        }

        int answer = Integer.MIN_VALUE;
        //don't consider the last portion
        for(int i = 0;i<s.length() - 1;i++) {
            if(s.charAt(i) == '0') {
                left_zero++; //1 2
                right_zero--;//1 0
            } else {
                right_one--;//3 2 1
                left_one++;// 1 2 3
            }
            
            answer = Math.max(answer, left_zero + right_one);
        }

        return answer;
    }
}